### PR TITLE
👤 Add Users

### LINKED ISSUE #133: ✨ Add Users
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Suggestion

Ability to add users to the system including first name, last name, and e-mail

### Possible Implementation

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by lprichar**
### Issue

Fixes #133 

### Description
Adds the ability to view and add users.

### Testing
1. Log in
2. Click Manage Users

Expected: you should see a list of users

image

3. Click the add button in the upper-right

Expected: you should navigate to the add users page

image

4. Enter in valid information and hit submit

Expected: You should navigate back to the manage users screen where your new user should be listed

### CODE REVIEW

File: packages/admin-app/src/components/AssignmentTable/AssignmentTable.tsx [hunk 1]
```diff
@@ -56,7 +58,7 @@ export const AssignmentTable: React.FC<AssignmentTableProps> = ({
                         }}
                         hideFooterPagination
                         selectionModel={selectionModel}
-                        checkboxSelection
+                        checkboxSelection={!!onChanged}
```
Comment (lprichar) [inline packages/admin-app/src/components/AssignmentTable/AssignmentTable.tsx:61]: In other words AssignmentTable was previously for selecting users, but now it has been repurposed to just display users

File: packages/admin-app/src/components/ErrorMessage/ErrorMessage.tsx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+export interface ErrorMessageProps extends AlertProps {
+    MessageId: string;
+}
+
+export const ErrorMessage: React.FC<ErrorMessageProps> = (props) => {
+    const { MessageId } = props;
+    const alertProps = props as AlertProps;
+
+    return (
+        // eslint-disable-next-line react/jsx-props-no-spreading
```
Comment (lprichar) [inline packages/admin-app/src/components/ErrorMessage/ErrorMessage.tsx:14]: This eslint disabling is used 4 times throughout the project, should we maybe just disable it globally?

File: packages/admin-app/src/pages/UserManagementPage.tsx [hunk 3]
```diff
@@ -1,6 +1,59 @@
...
+    return (
+        <Container maxWidth="md" className={classes.content}>
+            <Box>
+                <Grid container alignItems="center">
+                    <Grid item xs={1} />
+                    <Grid item xs={10}>
+                        <IconHeader title={new Message(MessageId.UserManagement_Title)} />
+                    </Grid>
+                    <Grid item xs={1} className={classes.buttonArea}>
+                        <Fab href={routeIds.addUser} color="primary" size="medium" aria-label="add">
```
Comment (lprichar) [inline packages/admin-app/src/pages/UserManagementPage.tsx:48]: I don't love the UI for the add button.  I may return to it if I find something better.  I'm also open to suggestions.  I know Floating Action Buttons are common in material, so I tried floating it in the bottom right, but it just didn't feel quite right.  Here at the top of the table felt slightly better.  Good enough for now I guess.

File: packages/admin-app/src/components/ErrorMessage/ErrorMessage.tsx (hunk 2 above)
Comment (SteveMaier-IRT) [reply packages/admin-app/src/components/ErrorMessage/ErrorMessage.tsx:14]: So we are disabling it everytime we use the spreading?  If that is the case, I would agree to disable it.

Comment (lprichar) [reply packages/admin-app/src/components/ErrorMessage/ErrorMessage.tsx:14]: It's actually a little more nuanced than that.  It seems to be necessary and disabled in components that extend another component and need to enable consumers to use the original component's props plus an extra property or two.  In contexts outside of that scenario it's maybe still a good idea to disable prop spreading.  Unfortunately, I don't have enough experience yet to know how common that lint rule might prove useful.

**Review by SteveMaier-IRT — APPROVED** [review]
Looks good.

**Pull request merged**